### DISCUSSION TITLE
Script to play MP3 file on_event_start

### DISCUSSION THREAD

**Discussion by tqchiem** [Motion-Project/motion#1779] [General]
I am running Motion 4.3.2 on Ubuntu 22.04 LTS and would like to play an mp3 file "on_event_start" using "mpg123" audio player via a simple bash script (myscript.sh) below:

```
#!/bin/bash
mpg123 myaudio.mp3
```
This script runs fine in the terminal (like 'bash myscript.sh') but it does not on the 'on_event_start' trigger from Motion. It has permission error, such as below:

```
Jul 03 03:36:53 ubuntu motion[2175]: AL lib: (EE) ALCplaybackAlsa_open: Could not open playback device 'default': No such file or directory
Jul 03 03:36:53 ubuntu motion[2175]: [src/libmpg123/readers.c:INT123_open_stream():1204] error: Cannot open file /home/ubuntu/myaudio.mp3: Permission denied
Jul 03 03:36:53 ubuntu motion[2175]: main: [src/mpg123.c:open_track():776] error: Cannot open /home/ubuntu/myaudio.mp3: File access error. (code 22)
```
To see what happens after that, I did chmod 755 and chown motion:motion on myaudio.mp3 and myscript.sh...the same errors.

Any help/hint would be greatly appreciated. Thanks.

BTW...what is the motion account default password?.

**Comment by tosiara** [comment]
Motion process is running as `motion` user, and your mp3 is located in user's `ubuntu` home dir. Try to put it into `/etc` and change script to:
```
#!/bin/bash
mpg123 /etc/myaudio.mp3
```

**Comment by tqchiem** [comment]
Thanks for your suggestion. 
I put the audio file in /etc/motion and updated the script to:

```
#!/bin/bash
mpg123 /etc/motion/myaudio.mp3
```
This does not work, but it no longer has those permission errors. 

After trial and error, the configuration that works for me is to run mpg123 with sudo in the script AND add "motion ALL=(ALL) NOPASSWD:ALL" to the end of the "/etc/sudoers" file. With this, the audio file can be in any directory.

I use "on_event_start" to trigger the script. When I wave my hand in front of the camera, it seems motion runs two instances of the script. The audio plays over one another with very small delay, but noticeable. I can see two copies of audio playing via journalctl. Any idea as to why?. Also, what is the maximum event number that motion uses to stamp the media filename before it resets?. After restarted, motion starts with 01.

Thanks

**Comment by tqchiem** [comment]
After looking into it, it's my own doing for having two instances of the audio playing. Motion works correctly. I have 4 cameras and I set on_event_start in motion.conf. In my case, two cameras see the same event and therefore trigger the script. I moved on_event_start to individual cameraX.conf to avoid this problem. I still need to know the max event ID.

**Reply by tosiara** [reply]
Event id is `int`: https://github.com/Motion-Project/motion/blob/master/src/motion.h#L430
So, max would be `2147483647`
